packages/analyzer/src/javascript/class-scanner.ts

```diff
@@ -182,6 +186,62 @@ export class ClassScanner implements JavaScriptScanner {
     };
   }
 
+  /**
+   * Handle the pattern where a class's superclass is declared as a separate
+   * variable, usually so that mixins can be applied in a way that is compatible
+   * with the Closure compiler.
+   */
+  private collapseEphemeralSuperclasses(classes: ClassType[]): ClassType[] {
+    const classById = new Map<string, ClassType>();
+    const classBySuperClassId = new Map<string, ClassType[]>();
+
+    for (const cls of classes) {
+      if (cls.name === undefined) {
+        continue;
+      }
+      classById.set(cls.name, cls);
+
+      if (cls.superClass !== undefined) {
+        const superClassId = cls.superClass.identifier;
+        const childClasses = classBySuperClassId.get(superClassId);
+        if (childClasses === undefined) {
+          classBySuperClassId.set(superClassId, [cls]);
+        } else {
+          childClasses.push(cls);
+        }
+      }
+    }
+
+    const ephemeralSuperClasses = new Set<ClassType>();
+    for (const [superClassId, childClasses] of classBySuperClassId) {
+      const superClass = classById.get(superClassId);
+      if (superClass === undefined) {
+        continue;
+      }
+      if (superClass.privacy !== 'private') {
+        continue;
+      }
+      let isEphemeral = false;
+      for (const childClass of childClasses) {
+        if (superClass.superClass !== undefined) {
+          // Feature properties are readonly, hence this hacky cast. We could
+          // also make a new feature, but then we'd need a good way to clone a
+          // feature.
```

Comment:
Maybe mention that this is ok because this is still part of the "construction" phase of the scanned classes, they're entirely file local, and they haven't left the ClassScanner yet.

Reply:
Done.